tsdb/wal/watcher.go

```diff
@@ -496,18 +533,21 @@ func (w *Watcher) readSegment(r *LiveReader, segmentNum int, tail bool) error {
 				return err
 			}
 			for _, s := range samples {
+				w.samplesRead++
 				if s.T > w.startTimestamp {
```

Comment:
Hmm, doesn't this have the same problem? Even if we start at a newer segment, we're still not going to queue any new samples until after time.Now() has passed.

Reply:
It's smelly but at the moment we're only setting the start time if we didn't find/set the starting segment.
```
if startSegment == -1 {
	w.SetStartTime(time.Now())
}
```